src/api/apps/articles/test/model/retrieve.spec.js

```diff
@@ -0,0 +1,158 @@
+import { toQuery } from "../../model/retrieve"
+
+let _ = require("underscore")
+const { fabricate, empty } = require("../../../../test/helpers/db")
+const gravity = require("antigravity").server
+const app = require("express")()
+_ = require("underscore")
+const { ObjectId } = require("mongojs")
+
+describe("Retrieve", function() {
+  let server
+  before(function(done) {
+    app.use("/__gravity", gravity)
+    server = app.listen(5000, () => done())
+    return server
+  })
+
+  after(function() {
+    return server.close()
+  })
+
+  beforeEach(done =>
+    empty(() => fabricate("articles", _.times(10, () => ({})), () => done())))
+
+  describe("#toQuery", function() {
+    it("aggregates the query for all_by_author", function() {
+      const { query } = toQuery({
+        all_by_author: ObjectId("5086df098523e60002000017"),
+        published: true,
+      })
+      query["$or"][0].author_id.should.containEql(
+        ObjectId("5086df098523e60002000017")
+      )
+      query["$or"][1].contributing_authors["$elemMatch"].should.be.ok()
+      query["$or"][1].contributing_authors["$elemMatch"].id.should.containEql(
+        ObjectId("5086df098523e60002000017")
+      )
+    })
+
+    it("aggregates the query for vertical", function() {
+      const { query } = toQuery({
+        vertical: "55356a9deca560a0137bb4a7",
+        published: true,
+      })
+      query["vertical.id"].should.containEql(
```

Comment:
To switch to `jest` syntax use expect rather than should, 
`expect(query["vertical.id"]).toBe(ObjectId("55356a9deca560a0137bb4a7"))`
https://jestjs.io/docs/en/expect